apps/meteor/app/lib/server/lib/validateCustomMessageFields.ts

```diff
@@ -0,0 +1,37 @@
+import Ajv from 'ajv';
+import mem from 'mem';
+
+const ajv = new Ajv();
+
+const customFieldsValidate = mem((customFieldsSetting: string) => {
+	const schema = JSON.parse(customFieldsSetting);
+
+	if (schema.type && schema.type !== 'object') {
+		throw new Error('Invalid custom fields config');
+	}
+
+	return ajv.compile({
+		...schema,
+		type: 'object',
+		additionalProperties: false,
+	});
+});
+
+export const validateCustomMessageFields = (
+	customFields: Record<string, any>,
+	messageCustomFieldsEnabled: boolean,
```

Comment:
I think it would be better to provide an object for params to avoid this boolean param, it's clearer to understand. WDYT?

Reply:
Done